src/main/java/net/minestom/server/item/ItemMetaBuilder.java

```diff
@@ -0,0 +1,207 @@
+        });
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder lore(@NotNull List<@NotNull Component> lore) {
+        this.lore = lore;
+        handleCompound("display", nbtCompound -> {
+            final NBTList<NBTString> loreNBT = new NBTList<>(NBTTypes.TAG_String);
+            for (Component line : lore) {
+                loreNBT.add(new NBTString(GsonComponentSerializer.gson().serialize(line)));
+            }
+            nbtCompound.set("Lore", loreNBT);
+        });
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder lore(Component... lore) {
+        lore(Arrays.asList(lore));
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder enchantments(@NotNull Map<Enchantment, Short> enchantments) {
+        this.enchantmentMap = enchantments;
+
+        if (!enchantmentMap.isEmpty()) {
+            NBTUtils.writeEnchant(nbt, "Enchantments", enchantmentMap);
+        } else {
+            this.nbt.removeTag("Enchantments");
+        }
+
+        return this;
+    }
+
+    @Contract("_, _ -> this")
+    public @NotNull ItemMetaBuilder enchantment(@NotNull Enchantment enchantment, short level) {
+        this.enchantmentMap.put(enchantment, level);
+        enchantments(enchantmentMap);
+        return this;
+    }
+
+    @Contract("-> this")
+    public @NotNull ItemMetaBuilder clearEnchantment() {
+        this.enchantmentMap.clear();
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder attributes(@NotNull List<@NotNull ItemAttribute> attributes) {
+        this.attributes = attributes;
+
+
+        if (!attributes.isEmpty()) {
+            NBTList<NBTCompound> attributesNBT = new NBTList<>(NBTTypes.TAG_Compound);
+
+            for (ItemAttribute itemAttribute : attributes) {
+                final UUID uuid = itemAttribute.getUuid();
+                attributesNBT.add(
+                        new NBTCompound()
+                                .setIntArray("UUID", Utils.uuidToIntArray(uuid))
+                                .setDouble("Amount", itemAttribute.getValue())
+                                .setString("Slot", itemAttribute.getSlot().name().toLowerCase())
+                                .setString("AttributeName", itemAttribute.getAttribute().getKey())
+                                .setInt("Operation", itemAttribute.getOperation().getId())
+                                .setString("Name", itemAttribute.getInternalName())
+                );
+            }
+            this.nbt.set("AttributeModifiers", attributesNBT);
+        } else {
+            this.nbt.removeTag("AttributeModifiers");
+        }
+
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder customModelData(int customModelData) {
+        this.customModelData = customModelData;
+        this.nbt.setInt("CustomModelData", customModelData);
```

Comment:
Is modifying the NBT a relatively costly operation? Might be better to just create the NBT once the builder is built if so.